### DISCUSSION TITLE
How does Mason find python?

### DISCUSSION THREAD

**Discussion by smprather** [mason-org/mason.nvim#1583] [Q&A]
Mason is latching onto a python install that is not the one I want. I am setting the one I want in my PATH, then launching nvim. But :checkhealth shows that Mason is not find the one in my PATH. I do not have g:python3_host_prog set.

I am working from a clean git-clone of kickstart.nvim.

In case it matters, I'm using NVIM_APPNAME to target my kickstart.nvim install.

Figured it out. I had changed my shell env PATH var, but I had not changed it in my .bashrc. Looks like Mason is using login shells to find stuff. I'm ok to move forward, but this doesn't seem like the right thing for Mason to do.

```
$ type -a python3
python3 is /proj/cot_globals/tools/foss/python/3.12.1/bin/python3
python3 is /proj/cot_globals/envs/globals_python3.11/bin/python3
python3 is /proj/cot_cad/conda/envs/cot_glb_py0/bin/python3
python3 is /usr/bin/python3
python3 is /bin/python3

:checkhealth
mason.nvim [Languages]
OK python: Python 3.11.0rc1
OK python venv: Ok
OK pip: pip 22.0.2 from /proj/cot_globals/envs/globals_python3.11/lib/python3.11/site-packages/pip (python 3.11)

:echo g:python3_host_prog
E121: Undefined variable: g:python3_host_prog

**Comment by ashemedai** [comment]
A very delayed answer, but I ran into this myself today, wondering why it kept using 3.12 when 3.14 is my latest installed version.

Well, it's hardcoded to try a bunch of different interpreter versions, with the maximum version being 3.12: https://github.com/mason-org/mason.nvim/blob/ad7146aa61dcaeb54fa900144d768f040090bff0/lua/mason-core/installer/managers/pypi.lua#L60-L77
